src/extension.ts

```diff
@@ -1529,6 +1531,78 @@ export async function activate(context: vscode.ExtensionContext) {
         wsServer.close();
       });
   });
+  registerIDFCommand("espIdf.jtag_flash", () => {
+    PreCheck.perform([openFolderCheck], async () => {
+      const buildFolder = path.join(workspaceRoot.fsPath, "build");
+      if (!existsSync(buildFolder)) {
```

Comment:
We should also check `.elf` is generated, build folder is still created on build failure.

Reply:
Wouldn't be better to use `pathExists` from path-extra for async instead of fs `existsSync` ?